### PR TITLE
refactor: instantiate contracts via Core SDK

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Using the Core SDK to get contract instances.

## How this PR fixes it

All (possible) contract instance retrievals are now done via the Core SDK so that we rely less on TypeChain.

## How to test it

- Safe creation and associated gas estimation works as expected.
- Interrupted Safe creation transactions succeed.
- Batch execution works as expected.
- Safe updating works as expected.

### CODE REVIEW

File: src/components/tx/modals/BatchExecuteModal/ReviewBatchExecute.tsx [hunk 1]
```diff
@@ -65,61 +65,59 @@ const ReviewBatchExecute = ({ data, onSubmit }: { data: BatchExecuteData; onSubm
   const submitDisabled = loading || !isSubmittable
 
   return (
-    <div>
```
Comment (usame-algan) [inline src/components/tx/modals/BatchExecuteModal/ReviewBatchExecute.tsx:68]: This wrapper is needed with TxModal for the spacing. We could refactor this separately though.
Screenshot 2022-09-02 at 09 43 15

**Review by usame-algan — CHANGES_REQUESTED** [review]
There is a missing dependency. Should we also remove the typechain files and configs now that the contracts are not used anymore?

**Comment by iamacook** [comment]
@usame-algan, I've restored the wrapper, as well as added the missing dependency.

Not all contract instances are retrieved from the SDK yet. I think we should leave it as is for now.

**Review by usame-algan — APPROVED** [review]

**Comment by francovenica** [comment]
I've tried several tx in different situations like the ones mentioned in the description and they worked fine. 
I'll keep checking tx during the next week, if any issue is found it will be reported in a separed ticket

**Pull request merged**